pkgs/operator/operator.go

```diff
@@ -139,47 +132,91 @@ func RegisterRoutes(s *Server) {
 			rawdata, _ := io.ReadAll(request.Body)
 			signedReshareMsg := &wire.SignedTransport{}
 			if err := signedReshareMsg.UnmarshalSSZ(rawdata); err != nil {
-				s.Logger.Error("parsing failed: ", zap.Error(err))
-				writer.WriteHeader(http.StatusBadRequest)
-				writer.Write(wire.MakeErr(err))
+				utils.WriteErrorResponse(s.Logger, writer, err, http.StatusBadRequest)
 				return
 			}
 
 			// Validate that incoming message is an init message
 			if signedReshareMsg.Message.Type != wire.ReshareMessageType {
-				s.Logger.Error("received bad msg non init message sent to init route")
-				writer.WriteHeader(http.StatusBadRequest)
-				writer.Write(wire.MakeErr(errors.New("not init message to init route")))
+				utils.WriteErrorResponse(s.Logger, writer, errors.New("not init message to init route"), http.StatusBadRequest)
 				return
 			}
 			reqid := signedReshareMsg.Message.Identifier
 			logger := s.Logger.With(zap.String("reqid", hex.EncodeToString(reqid[:])))
 			logger.Debug("initiating instance with init data")
 			b, err := s.State.InitInstanceReshare(reqid, signedReshareMsg.Message, signedReshareMsg.Signature)
 			if err != nil {
-				logger.Error(fmt.Sprintf("failed to initiate instance err:%v", err))
-
-				writer.WriteHeader(http.StatusBadRequest)
-				writer.Write(wire.MakeErr(err))
+				utils.WriteErrorResponse(s.Logger, writer, err, http.StatusBadRequest)
 				return
 			}
 			logger.Info("✅ Instance started successfully")
 
 			writer.WriteHeader(http.StatusOK)
 			writer.Write(b)
 		})
+		s.Router.Route("/health_check", func(r chi.Router) {
+			r.Post("/", func(writer http.ResponseWriter, request *http.Request) {
+				rawdata, _ := io.ReadAll(request.Body)
+				signedPintMsg := &wire.SignedTransport{}
+				if err := signedPintMsg.UnmarshalSSZ(rawdata); err != nil {
+					utils.WriteErrorResponse(s.Logger, writer, err, http.StatusBadRequest)
+					return
+				}
+				// Validate that incoming message is a ping message
+				if signedPintMsg.Message.Type != wire.PingMessageType {
+					utils.WriteErrorResponse(s.Logger, writer, errors.New("not ping message to ping route"), http.StatusBadRequest)
+					return
+				}
+				s.Logger.Debug("received a health check message")
+				b, err := s.State.Pong(signedPintMsg)
+				if err != nil {
+					utils.WriteErrorResponse(s.Logger, writer, err, http.StatusBadRequest)
+					return
+				}
+				writer.WriteHeader(http.StatusOK)
+				writer.Write(b)
+			})
+		})
+		s.Router.Route("/results", func(r chi.Router) {
```

Comment:
dont like this route, operator should be the one providing the results to initiator. all operators should be able to make up the results and sign independently.

Reply:
each operator has only his encrypted share, all enc shares combined at initiator to one keyshares file. We would need a round of exchanges between operators, or a simple message from initiator

Reply:
same with deposit data, we have a master sig which is a combination of partial sigs. No operator has all partial sigs at the end of the initial ceremony.

Reply:
why do they need them though? this data is needed for the initiator to deposit and register the validator to ssv. the operators don't need it. or else I'm missing something. 
@lior-blox / @MatusKysel  - 
do we want operators to have the complete data? isn't that a security concern? shouldn't it belong only to the initiator?